test/parallel/test-stream-toArray.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const common = require('../common');
+const {
+  Readable,
+} = require('stream');
+const assert = require('assert');
+const { setTimeout } = require('timers/promises');
+
+{
+  // Works on a synchronous stream
+  (async () => {
+    const tests = [
+      [],
+      [1],
+      [1, 2, 3],
+      Array(100).fill().map((_, i) => i),
```

Comment:
nit: you can avoid creating intermediary arrays (and imho it's easier to read, but feel free to disagree)
```suggestion
      Array.from({ length: 100 }, (_, i) => i),
```